snapcraft/plugins/catkin.py

```diff
@@ -170,9 +170,12 @@ def _find_package_deps(self):
                     self.builddir, 'src', pkg, 'package.xml')
                 with open(filename, 'r') as f:
                     self._deps_from_packagesxml(f, pkg)
-            except os.FileNotFound:
-                logger.warning('Unable to find packages.xml for "' + pkg + '"')
-                pass
+            except IOError as e:
+                if e.errno is os.errno.ENOENT:
+                    logger.warning(
+                        'Unable to find packages.xml for "{}"'.format(pkg))
+                else:
+                    raise e
```

Comment:
This last branch is not tested. What can cause this method to raise an exception?